### PR TITLE
Replace Carthage support with bazel support

### PR DESCRIPTION
**Pull request by jverkoey**
bazel is much faster and doesn't require any tooling installations or clones because it's supported by kokoro.

Note that this PR deletes the xcodeproj that was created for Carthage support and introduces a BUILD file.

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        "-Wmissing-prototypes",  # Global function is defined without a previous prototype.
+        "-Wno-error=deprecated",  # Deprecation warnings are never errors.
+        "-Wno-error=deprecated-implementations",  # Deprecation warnings are never errors.
+        "-Wno-sign-conversion",  # Do not warn on sign conversions.
+        "-Wno-unused-parameter",  # Do not warn on unused parameters.
+        "-Woverlength-strings",  # Strings longer than the C maximum.
+        "-Wshadow",  # Local variable shadows another variable, parameter, etc.
+        "-Wstrict-selector-match",  # Compiler can't figure out the right selector.
+        "-Wundeclared-selector",  # Compiler doesn't see a selector.
+        "-Wunreachable-code",  # Code will never be reached.
```
Comment (jverkoey) [inline BUILD:37]: Am wondering if we can share these warnings somewhere? Perhaps as a separate bazel config file in foundation?

@ajsecord @randallli  thoughts?

**Comment by ianegordon** [comment]
Is anyone external using Carthage with Catalog-by-Convention?
I've been seeing more requests to add Carthage recently, if it doesn't cost much it may be worth keeping.

**Comment by jverkoey** [comment]
It unfortunately costs quite a bit to maintain because it requires that we check in xcodeproj files. We will then need to maintain these project files alongside our Podfile and BUILD file (which we already maintain internally anyway).

I think it's not worth the engineering overhead because its primary benefit is build performance, but bazel is much faster.

**Comment by jverkoey** [comment]
Reviewers: this is ready for review. Once this lands I'll move forward with deploying bazel + kokoro on the material motion repos.

**Comment by jverkoey** [comment]
Gentle ping.

File: BUILD (hunk 1 above)
Comment (jverkoey) [reply BUILD:37]: I'll probably explore pulling this out in a follow-up change.

Comment (randallli) [reply BUILD:37]: sgtm

**Review by randallli — APPROVED** [review]

**Pull request merged**